ape_solidity/compiler.py

```diff
@@ -664,6 +664,19 @@ def enrich_error(self, err: ContractLogicError) -> ContractLogicError:
         if not is_0x_prefixed(err.revert_message):
             return err
 
+        # ape-hardhat will deliver panic codes directly (no tracing needed)
+        revert_int = int(err.revert_message, 16)
+        if revert_int in [t.value for t in RuntimeErrorType]:
+            error_type = RuntimeErrorType(revert_int)
+            error_cls = RUNTIME_ERROR_MAP[error_type]
```

Comment:
If `RuntimeErrorType` subclasses `enum.IntEnum`, we can simplify as follows:
```suggestion
        error_type = int(err.revert_message, 16)
        if error_type in iter(RuntimeErrorType):
            error_cls = RUNTIME_ERROR_MAP[error_type]
```

Reply:
I really dont like IntEnum... I think it is abuse of Python personally. but we can if you insist.

Reply:
I heard they were going to remove IntEnum though.

Reply:
> I heard they were going to remove IntEnum though.

If this is the case, disregard

Although I think what they meant is to do `class IntegerEnum(int, Enum)` instead

Reply:
no, it is just int enum does not do everything right that ints do, so it gets weird in some cases and i always forget what those are, so in general easier to avoid and wrap in int() when needing it to be an int and enum when needing to be enum.

i did make the change tho. we use IntEnums elsewhere and in this case. Error codes are the best int enums probably.